uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Import/Parser/BaseClass.php

```diff
@@ -0,0 +1,323 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRLeaveAndAbsences_BAO_AbsenceType as AbsenceType;
+
+/**
+ * Class to parse activity csv files.
+ */
+class CRM_HRLeaveAndAbsences_Import_Parser_BaseClass extends CRM_HRLeaveAndAbsences_Import_Parser {
+
+  use CRM_HRLeaveAndAbsences_Import_Helpers_DataImportTrait;
+
+  /**
+   * @var array
+   */
+  protected $_mapperKeys;
+
+  /**
+   * @var array
+   */
+  private $absenceTypes;
+
+  /**
+   * @var array
+   */
+  private $absenceStatuses;
+
+  /**
+   * @var array
+   */
+  private $leaveImportSuccess = [];
+
+  /**
+   * @var array
+   */
+  private $leaveImportError = [];
+
+  /**
+   * Class constructor.
+   *
+   * @param array $mapperKeys
+   */
+  public function __construct(&$mapperKeys) {
+    parent::__construct();
+    $this->_mapperKeys = &$mapperKeys;
+  }
+
+  /**
+   * Function of undocumented functionality required by the interface.
+   */
+  protected function fini() {}
+
+  /**
+   * The initializer code, called before the processing.
+   */
+  public function init() {
+    $activityContact = CRM_Activity_BAO_ActivityContact::import();
+    $activityTarget['target_contact_id'] = $activityContact['contact_id'];
```

Comment:
You're not using these variables anywhere else in this method. Do you really need them?